### PR TITLE
SW-2629 Set id to all onChange() functions

### PR DESCRIPTION
**Pull request by constanzauanini**
This is because new version of web-component doesn't return the id from the component, so we need to explicitly set it. After this change any component's id can be changed to whatever id we need in the HTML, because data model id is set in the onChange function

### CODE REVIEW

File: src/components/Inventory/view/BatchDetailsModal.tsx
Comment (karthikbtf) [inline src/components/Inventory/view/BatchDetailsModal.tsx:196]: typo `seedlingsBatch` .. guessing we don't need the onChange here and in the other places - since it's readonly?

File: src/components/Inventory/view/WithdrawalModal.tsx
Comment (karthikbtf) [inline src/components/Inventory/view/WithdrawalModal.tsx:201]: same typo,, and remove onChange?

File: src/components/NurseryWithdrawals/ReassignmentRenderer.tsx
Comment (karthikbtf) [inline src/components/NurseryWithdrawals/ReassignmentRenderer.tsx:100]: we can remove the id here, it is unused,, also maybe remove it from onUpdateQuantity

**Review by karthikbtf — COMMENTED** [review]
Thanks for doing this!!!
A few comments.
Also - my suggestion is we wait until tracking-v1 is released to production before we get this in. Gives us more time to test, since this is a risky change. Thoughts?

**Comment by constanzauanini** [comment]
> BatchDetailsModal.tsx

Sure, I'll merge this after tracking v1 is released

File: src/components/Inventory/view/BatchDetailsModal.tsx
Comment (karthikbtf) [inline src/components/Inventory/view/BatchDetailsModal.tsx:192]: seedlingsBatch?

**Review by karthikbtf — APPROVED** [review]
Thanks for the changes, looks good.
Let's plan how to test these changes before we merge it in? We can maybe share the load.. and merge after the production release.

**Comment by constanzauanini** [comment]
> Thanks for the changes, looks good. Let's plan how to test these changes before we merge it in? We can maybe share the load.. and merge after the production release.

We need to test all these flows:

- Create new organization
- Edit organization
- Create inventory
- Filter inventory list
- Search inventory list
- Add new batch
- Edit batch
- Single seedling withdrawal 
- Multiple seedling withdrawal
- Sensor kit set up flow
- Edit my account
- Create nursery
- Edit nursery
- Search seedbank
- Create seedbank
- Edit seedbank
- Search withdrawal log
- Filter withdrawal log
- Reassignment flow
- Search people
- Create person
- Edit person
- Search planting site
- Create planting site
- Edit planting site
- Plant dashboard (change zone and plot)
- Search species
- Filter Species
- Create new species
- Edit species
- Search accession
- Filter accession
- Create accession
- Edit accession
- Add / edit viability tests
- Accession withdraw flow

**Comment by constanzauanini** [comment]
This PR with the version that reverts mui/styles change in web-components, doesn't initialize the theme provider correctly. So I would suggest that we merge https://github.com/terraware/terraware-web/pull/987 directly and close this PR

**Comment by karthikbtf** [comment]
> This PR with the version that reverts mui/styles change in web-components, doesn't initialize the theme provider correctly. So I would suggest that we merge #987 directly and close this PR

@constanzauanini are you referring to this PR? https://github.com/terraware/web-components/pull/234/files
It does not break anything. It reverted the changes back to how things were when it worked before. I also tested it then and just now.
Ideally we should have just one PR with the onChange related changes.

**Comment by constanzauanini** [comment]
> > This PR with the version that reverts mui/styles change in web-components, doesn't initialize the theme provider correctly. So I would suggest that we merge #987 directly and close this PR
> 
> @constanzauanini are you referring to this PR? https://github.com/terraware/web-components/pull/234/files It does not break anything. It reverted the changes back to how things were when it worked before. I also tested it then and just now. Ideally we should have just one PR with the onChange related changes.

This PR only has the onChange related changes, but the Vercel link doesn't work and if you take a look at the console, the issue is related to the theme initialization.

**Comment by karthikbtf** [comment]
> > > This PR with the version that reverts mui/styles change in web-components, doesn't initialize the theme provider correctly. So I would suggest that we merge #987 directly and close this PR
> > 
> > 
> > @constanzauanini are you referring to this PR? https://github.com/terraware/web-components/pull/234/files It does not break anything. It reverted the changes back to how things were when it worked before. I also tested it then and just now. Ideally we should have just one PR with the onChange related changes.
> 
> This PR only has the onChange related changes, but the Vercel link doesn't work and if you take a look at the console, the issue is related to the theme initialization.

I suspect something else is off : see https://github.com/terraware/terraware-web/pull/990

**Comment by constanzauanini** [comment]
> > > > This PR with the version that reverts mui/styles change in web-components, doesn't initialize the theme provider correctly. So I would suggest that we merge #987 directly and close this PR
> > > 
> > > 
> > > @constanzauanini are you referring to this PR? https://github.com/terraware/web-components/pull/234/files It does not break anything. It reverted the changes back to how things were when it worked before. I also tested it then and just now. Ideally we should have just one PR with the onChange related changes.
> > 
> > 
> > This PR only has the onChange related changes, but the Vercel link doesn't work and if you take a look at the console, the issue is related to the theme initialization.
> 
> I suspect something else is off : see #990

Oh, thanks!! I hadn't seen that!

**Comment by karthikbtf** [comment]
> > > > > This PR with the version that reverts mui/styles change in web-components, doesn't initialize the theme provider correctly. So I would suggest that we merge #987 directly and close this PR
> > > > 
> > > > 
> > > > @constanzauanini are you referring to this PR? https://github.com/terraware/web-components/pull/234/files It does not break anything. It reverted the changes back to how things were when it worked before. I also tested it then and just now. Ideally we should have just one PR with the onChange related changes.
> > > 
> > > 
> > > This PR only has the onChange related changes, but the Vercel link doesn't work and if you take a look at the console, the issue is related to the theme initialization.
> > 
> > 
> > I suspect something else is off : see #990
> 
> Oh, thanks!! I hadn't seen that!

Just to confirm, I created a new PR https://github.com/terraware/terraware-web/pull/992 with only the version upgrade to web-components. If it fails, we can re-discuss the fix. If we want to go with version 57, we have to re-revert the revert :0 and create version 61 for Charlie to work off.

**Comment by constanzauanini** [comment]
> #992

Without this onChange changes, version 58 will fail because we need to remove the id parameters from the onChange. I just reverted package.json and yarn.lock to be like main and only changed the web-components version and still getting errors :(

**Comment by karthikbtf** [comment]
> > > > > > > This PR with the version that reverts mui/styles change in web-components, doesn't initialize the theme provider correctly. So I would suggest that we merge #987 directly and close this PR
> > > > > > 
> > > > > > 
> > > > > > @constanzauanini are you referring to this PR? https://github.com/terraware/web-components/pull/234/files It does not break anything. It reverted the changes back to how things were when it worked before. I also tested it then and just now. Ideally we should have just one PR with the onChange related changes.
> > > > > 
> > > > > 
> > > > > This PR only has the onChange related changes, but the Vercel link doesn't work and if you take a look at the console, the issue is related to the theme initialization.
> > > > 
> > > > 
> > > > I suspect something else is off : see #990
> > > 
> > > 
> > > Oh, thanks!! I hadn't seen that!
> > 
> > 
> > Just to confirm, I created a new PR #992 with only the version upgrade to web-components. If it fails, we can re-discuss the fix. If we want to go with version 57, we have to re-revert the revert :0 and create version 61 for Charlie to work off.
> 
> Without these onChange changes, version 58 will fail because we need to remove the id parameters from the onChange. I just reverted package.json and yarn.lock to be like main and only changed the web-components version and still getting errors :(

can you try one last thing.. create a new branch from this branch and send a PR out of that?

**Comment by constanzauanini** [comment]
PR is redone in https://github.com/terraware/terraware-web/pull/995

**Pull request closed without merging**
